packages/web-components/src/keplr-connection/KeplrConnection.js

```diff
@@ -12,17 +13,28 @@ const DEFAULT_NETWORK_CONFIG = 'https://main.agoric.net/network-config';
 export const makeAgoricKeplrConnection = async (
   networkConfig = DEFAULT_NETWORK_CONFIG,
   context = makeImportContext(),
+  chainStorageWatcher,
 ) => {
   const { chainId, rpcs } = await getChainInfo(networkConfig);
   const address = await getKeplrAddress(chainId);
 
-  const leader = makeLeader(networkConfig);
-  const walletNotifiers = await watchWallet(leader, address, context, rpcs);
+  let walletNotifiers;
+  let leader;
+  if (chainStorageWatcher) {
```

Comment:
in this case `leader` is undefined, which will break consumers depending on this value. So this is a breaking change that should have `!` in the commit message.

but while we're at it, please remove the `leader` property from this API. apparently it's not needed.

I imagine the intent was to re-use the leader throughout the client so it could optimize requests but as we don't have that yet a smaller API surface gives us more flexibility in how we design that. cc @michaelfig

Reply:
I don't think this is breaking because the behavior only changes if you use the new `chainStorageWatcher` param.